provisioning/miniprov/hare_mp/main.py

```diff
@@ -449,18 +470,19 @@ def init(args):
     try:
         url = args.config[0]
         utils = Utils(ConfStoreProvider(url))
-        stop_event = Event()
-        config_dir = get_config_dir(url)
-        log_dir = get_log_dir(url)
-        # Starting consul and hax
-        consul_starter = _start_consul(utils, stop_event,
-                                       config_dir, log_dir, url)
-        hax_starter = _start_hax(utils, stop_event, config_dir, log_dir)
-        hostname = utils.get_local_hostname()
-        start_mkfs_parallel(hostname, config_dir)
-        # Stopping hax and consul
-        hax_starter.stop()
-        consul_starter.stop()
+        if is_mkfs_required(url):
+           stop_event = Event()
+           config_dir = get_config_dir(url)
+           log_dir = get_log_dir(url)
+           # Starting consul and hax
+           consul_starter = _start_consul(utils, stop_event,
+                                          config_dir, log_dir, url)
+           hax_starter = _start_hax(utils, stop_event, config_dir, log_dir)
+           hostname = utils.get_local_hostname()
+           start_mkfs_parallel(hostname, config_dir)
+           # Stopping hax and consul
+           hax_starter.stop()
+           consul_starter.stop()
```

Comment:
[Optional]: if you invert the boolean expression, no nesting will be required (the code becomes simpler).

Update: while preparing this suggestion, I noticed that you use 3-space indentation. Please use 4 spaces as a single indent instead in your code.

```suggestion
        if not is_mkfs_required(url):
            return

        stop_event = Event()
        config_dir = get_config_dir(url)
        log_dir = get_log_dir(url)
        # Starting consul and hax
        consul_starter = _start_consul(utils, stop_event,
                                       config_dir, log_dir, url)
        hax_starter = _start_hax(utils, stop_event, config_dir, log_dir)
        hostname = utils.get_local_hostname()
        start_mkfs_parallel(hostname, config_dir)
        # Stopping hax and consul
        hax_starter.stop()
        consul_starter.stop()
```